tests/DocumentationExamplesTest.php

```diff
@@ -1486,6 +1480,12 @@ public function testCausalConsistency()
             [ 'sku' => '111', 'name' => 'Peanuts', 'start' => new UTCDateTime() ]
         );
 
+        try {
```

Comment:
Can you add a comment explaining why this was moved down from the top of the method (like we do for other skip conditions)? I think the long-term solution would be to create a new utility method that ensures our environment is a sharded cluster backed by multi-member replica sets, but we can punt on that if this is the only use case.

Reply:
Added:
> In sharded clusters, server selection ignores the read preference mode, so using $manager->selectServer does not work here. To work around this, we run a query on a secondary and rely on an exception to let us know that no secondary is available.